pkg/kubelet/status/manager.go

```diff
@@ -384,6 +397,40 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	glog.Warningf("Failed to update status for pod %q: %v", format.Pod(pod), err)
 }
 
+func (m *manager) reconcilePod(pod *api.Pod) {
```

Comment:
Can we split the function into two and move the first half into ReconcilePodStatus? 
 - ReconcilePodStatus should do the comparison against the internal cache (line 401-422)
 - reconcilePod does the rest apiStatusVersion check (line 423-line 432)